tests/test_group.rs

```diff
@@ -1,8 +1,125 @@
 use maelstrom::ciphersuite::*;
 use maelstrom::creds::*;
+use maelstrom::framing::*;
 use maelstrom::group::*;
 use maelstrom::key_packages::*;
 
+#[test]
+fn create_commit_optinal_path() {
+    let ciphersuite =
+        Ciphersuite::new(CiphersuiteName::MLS10_128_DHKEMX25519_AES128GCM_SHA256_Ed25519);
+    let group_aad = b"Alice's test group";
+
+    // Define identities
+    let alice_identity = Identity::new(ciphersuite, "Alice".into());
+    let bob_identity = Identity::new(ciphersuite, "Bob".into());
+    let charlie_identity = Identity::new(ciphersuite, "Charlie".into());
+
+    // Define credentials
+    let alice_credential = BasicCredential::from(&alice_identity);
+    let bob_credential = BasicCredential::from(&bob_identity);
+    let charlie_credential = BasicCredential::from(&bob_identity);
+
+    // Generate KeyPackages
+    let alice_key_package_bundle = KeyPackageBundle::new(
+        &ciphersuite,
+        &alice_identity.get_signature_key_pair().get_private_key(), // TODO: bad API, we shouldn't have to get the private key out here (this function shouldn't exist!)
+        Credential::Basic(alice_credential.clone()), // TODO: this consumes the credential!
+        None,
+    );
+    let alice_key_package = alice_key_package_bundle.get_key_package();
+    let bob_key_package_bundle = KeyPackageBundle::new(
+        &ciphersuite,
+        &bob_identity.get_signature_key_pair().get_private_key(), // TODO: bad API, we shouldn't have to get the private key out here (this function shouldn't exist!)
+        Credential::Basic(bob_credential), // TODO: this consumes the credential!
+        None,
+    );
+    let bob_key_package = bob_key_package_bundle.get_key_package();
+
+    let alice_update_key_package_bundle = KeyPackageBundle::new(
+        &ciphersuite,
+        &alice_identity.get_signature_key_pair().get_private_key(), // TODO: bad API, we shouldn't have to get the private key out here (this function shouldn't exist!)
+        Credential::Basic(alice_credential),
```

Comment:
Should this be a different credential? (Then we wouldn't have to clone it further up.)

Reply:
Not necessarily, a credential can be very long-term.